From a9e15197d9178ca216b41ef43e8c5e72338d6175 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Sat, 1 Oct 2011 10:32:40 +0200
Subject: [PATCH 2/3] Make it build

---
 src/AddressSpace.hpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/AddressSpace.hpp b/src/AddressSpace.hpp
index bebd3e6..b09dd61 100644
--- a/src/AddressSpace.hpp
+++ b/src/AddressSpace.hpp
@@ -256,6 +256,7 @@ LocalAddressSpace::getEncodedP(pint_t& addr, pint_t end, uint8_t encoding)
 
 inline bool LocalAddressSpace::findUnwindSections(pint_t addr, pint_t& mh, pint_t& dwarfStart, pint_t& dwarfLen, pint_t& compactStart)
 {
+#if 0
 	dyld_unwind_sections info;
 	if ( _dyld_find_unwind_sections((void*)addr, &info) ) {
 		mh				= (pint_t)info.mh;
@@ -264,12 +265,14 @@ inline bool LocalAddressSpace::findUnwindSections(pint_t addr, pint_t& mh, pint_
 		compactStart	= (pint_t)info.compact_unwind_section;
 		return true;
 	}
+#endif
 	return false;
 }
 
 
 inline bool	LocalAddressSpace::findFunctionName(pint_t addr, char* buf, size_t bufLen, unw_word_t* offset)
 {
+#if 0
 	dl_info dyldInfo;
 	if ( dladdr((void*)addr, &dyldInfo) ) {
 		if ( dyldInfo.dli_sname != NULL ) {
@@ -278,6 +281,7 @@ inline bool	LocalAddressSpace::findFunctionName(pint_t addr, char* buf, size_t b
 			return true;
 		}
 	}
+#endif
 	return false;
 }
 
-- 
1.7.1

