From d4a94956f3073e84a86afc16455204d7558d61f5 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Fri, 30 Sep 2011 20:04:19 +0200
Subject: [PATCH 3/9] Do not grok objc ABI in otool

---
 otool/main.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/otool/main.c b/otool/main.c
index d4bdaf1..455daa9 100644
--- a/otool/main.c
+++ b/otool/main.c
@@ -1369,6 +1369,7 @@ void *cookie) /* cookie is not used */
 			    loc_relocs, nloc_relocs, vflag);
 	    }
 	    else{
+#ifdef HAVE_OBJC_OBJC_RUNTIME_H
 		 /*
 		  * This is the 32-bit non-arm cputype case.  Which is normally
 		  * the first Objective-C ABI.  But it may be the case of a
@@ -1416,6 +1417,9 @@ void *cookie) /* cookie is not used */
 			    nsorted_symbols, ext_relocs, next_relocs,
 			    loc_relocs, nloc_relocs, vflag);
 		}
+#else
+		printf("Objective-C not supported\n");
+#endif
 	    }
 	}
 
-- 
1.7.1

