From 9c45a699b320e9ac3391ded2324249b327ee75f6 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 5 Oct 2011 07:41:52 +0200
Subject: [PATCH 9/9] Fix ranlib discovery

---
 libstuff/execute.c |    4 ++--
 misc/libtool.c     |    3 +++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/libstuff/execute.c b/libstuff/execute.c
index 3455eab..c553144 100644
--- a/libstuff/execute.c
+++ b/libstuff/execute.c
@@ -155,9 +155,9 @@ char *str)
 	 * Construct the prefix to the program running.
 	 */
 	p = buf;
-	readlink("/proc/self/exe", buf, sizeof(buf));
+	buf[readlink("/proc/self/exe", buf, sizeof(buf))] = '\0';
 	prefix = realpath(p, resolved_name);
-	p = rindex(prefix, '/');
+	p = rindex(prefix, '-');
 	if(p != NULL)
 	    p[1] = '\0';
 
diff --git a/misc/libtool.c b/misc/libtool.c
index dc43659..6753be9 100644
--- a/misc/libtool.c
+++ b/misc/libtool.c
@@ -357,12 +357,15 @@ char **envp)
 	(void)umask(oumask);
 
 	/* see if this is being run as ranlib */
+#if 0
 	p = strrchr(argv[0], '/');
 	if(p != NULL)
 	    p++;
 	else
 	    p = argv[0];
 	if(strncmp(p, "ranlib", sizeof("ranlib") - 1) == 0)
+#endif
+	if(strstr(argv[0], "ranlib"))
 	    cmd_flags.ranlib = TRUE;
 
 	/* The default is to used long names */
-- 
1.7.1

